docs/wfforward.md

```diff
@@ -0,0 +1,1117 @@
+
+Our mutation scheme of continuous mutation positions on a specific interval would require some care in order to properly implement back-mutation.  We will not bother to do so.  However, it is important to realize that it would be incorrect to consider the case where a new draw from `np.random.random_sample` is the same as a currently-existing mutation position as a back mutation.  The reason why this would be an error has to do with how floating-point values are distributed within specific intervals and how random number generators are implemented.  To do models with back mutation correctly:
+
+* Define genomic regions as integers of intervals, $R \in {0, 1, 2, \ldots, L-1}$, where $L$ is the region length.
+* Use functions designed to sample from ranges of integers to generate mutation positions.  For example, [`np.random.randint`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.random.randint.html).
+
+The introduction of mutations into the simulation creates a condition where the final output can depend on how often we simplify.  This differs from the case of simply simulating segregation and recombiantion events.  The reason is that the final results of a simululation can only be identical if the same random number seed is used and if all accesses to the random number generator (RNG) occur in the same order.  (For example, if you mutate then recombine, you'll get a different result from recombining before mutating.)  When simulating mutations using our infinitely-many sites scheme, the contents of `lookup` depends on the GC interval. For example, we may draw a new mutation position and reject it (because it is already in `lookup`) with one GC interval, but accept it with another GC interval (becuase that lineage is removed during simplification, making that position a candidate mutation target again). For models with reversible mutation, a simulation may as if a back mutation happens on a lineage that would not exist ("simplified out") with a different GC interval.  When things like this happen, the context of subsequent RNG accesses will differ, and thus node/edge/site/mutation data are very likely to differ. Note that this is not a bug, and results will be correct *in distribution* (well, should be--you do need to test that, too!).  Rather, it is a subtlety to keep in mind when implementing and testing simulations.
+
+(The fact that the above code raises no warnings is because we've done relatively few simulations with a small population size and a low mutation rate.)
+
+It is desirable, for testing purposes if nothing else, to have the final nodes/edges be invariant to the GC interval.  One way to do that, which I think is the easiest, is to make use of two RNG objects.  One is used for mutations, and the other for everything else.  Using numpy, we can accomplish this as follows:
+
+
+```python
+rng1 = np.random.RandomState(42)
+rng2 = np.random.RandomState(42)
+print(rng1.random_sample(2),rng2.random_sample(2))
```

Comment:
I'm slightly uneasy about this. Given that the seeds are the same, couldn't this add some subtle correlation between the genealogical and mutational processes? I don't see how, but my first instinct with RNG stuff is to be suspicious!